utils/advanced_stats_lib.py

```
@@ -0,0 +1,129 @@
+import statsmodels.stats.multitest as smt
+from statsmodels.stats.outliers_influence import variance_inflation_factor
+from patsy import dmatrix
+import logging
+
+logger = logging.getLogger(__name__)
+
+# --- Multiple Comparison Corrections (MCC) ---
+
+from typing import Union
+
+def apply_mcc(p_values: Union[list, pd.Series, np.ndarray], method: str = 'fdr_bh', alpha: float = 0.05) -> pd.Series:
```

Ruff (0.14.10)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)